soroban-sdk/src/env.rs

```diff
@@ -296,9 +283,27 @@ impl Env {
 
         let rf = Rc::new(EmptySnapshotSource());
         let storage = internal::storage::Storage::with_recording_footprint(rf);
-        Env {
-            env_impl: internal::EnvImpl::with_storage(storage),
-        }
+
+        internal::EnvImpl::with_storage(storage)
+    }
+
+    fn with_empty_recording_storage() -> Env {
+        let env_impl = Self::with_empty_recording_storage_env_impl();
+        let l = LedgerInfo {
+            protocol_version: 0,
+            sequence_number: 0,
+            timestamp: 0,
+            network_id: vec![0u8],
+        };
+        env_impl.set_ledger_info(l);
+        Env { env_impl }
+    }
+
+    /// Creates an [Env] with empty storage and a user specified [LedgerInfo]
+    pub fn with_empty_recording_storage_and_ledger_info(li: LedgerInfo) -> Env {
+        let env_impl = Self::with_empty_recording_storage_env_impl();
+        env_impl.set_ledger_info(li);
```

Comment:
The layering of these functions is not going to scale well to us setting other attributes on `Env` in tests.

Can we add a testutils function `set_ledger` (drop the `_info`) to `Env` instead? So the idea being someone will do this in a test:
```rust
let env = Env::default();
env.set_ledger(LedgerInfo{ ... });
```

Reply:
Done